libs/langgraph/langgraph/pregel/executor.py

```diff
@@ -34,41 +35,63 @@ def __call__(
         ...
 
 
-@contextmanager
-def BackgroundExecutor(config: RunnableConfig) -> Iterator[Submit]:
-    tasks: dict[concurrent.futures.Future, bool] = {}
-    with get_executor_for_config(config) as executor:
+class BackgroundExecutor(ContextManager):
```

Comment:
this rewrites BackgroundExecutor to wait all tasks before exit, like the async one